packages/TorneloScoresheet/src/pages/GraphicalRecording/MoveOptionsSheet.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import Sheet from '../../components/Sheet/Sheet';
+import { colours } from '../../style/colour';
+import { ICON_CLOCK, ICON_HALF } from '../../style/images';
+import { PlayerColour } from '../../types/ChessGameInfo';
+import MoveOption from './MoveOption';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+import { useGraphicalRecordingState } from '../../context/AppModeStateContext';
+
+export type EditingMove = {
+  colour: PlayerColour;
+  moveIndex: number;
+};
+
+type MoveOptionsSheetProps = {
+  editingMove: EditingMove | undefined;
+  dismiss: () => void;
+};
+
+const MoveOptionsSheet = ({ editingMove, dismiss }: MoveOptionsSheetProps) => {
+  const recordingState = useGraphicalRecordingState();
+  const actions = recordingState?.[1];
```

Comment:
I feel like this function should be passed through by the graphical recording state, so all the functions are declared in one spot, ie we shouldnt be accesing the state hook here

Reply:
Especially since the draw function is already defined in the graphical recording page, so if there is extra logic before calling the hook we need to change it in 2 places

Reply:
I'm a bit confused by this, they are all defined in the one spot : bottom of `AppModeStateContext.tsx`. This component will be using a few of the functionalities provided by `useGraphicalRecordingState`: The record time function, the draw offer function and the edit move function. The point of putting this stuff in the hook is so that it's decoupled from the component logic and can basically act as a "service" to the UI logic. I'm not saying your wrong, I'm just confused by what you're saying, so I'm happy to have a chat about this over zoom or something to clarify if you want?

Reply:
What i mean is that maybe there is extra logic that needs to be performed before calling the hook, which will need to be implemented both in GraphicalRecording page ( for the action buttons) and this component, so it might be better to pass through a function into this component that is used both by the action buttons in GraphicalRecording and this popup. But actually right now the toggle draw code is just 1 line of code so we can just go with your implementation for now